packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -541,12 +553,35 @@ class _CupertinoTextFieldState extends State<CupertinoTextField> with AutomaticK
   }
 
   void _handleSingleLongTapEnd(LongPressEndDetails details) {
-    _editableTextKey.currentState.showToolbar();
+    if (_isSelectionOverlayEnabled)
+      _editableText.showToolbar();
   }
 
   void _handleDoubleTapDown(TapDownDetails details) {
     _renderEditable.selectWord(cause: SelectionChangedCause.tap);
-    _editableTextKey.currentState.showToolbar();
+    if (_isSelectionOverlayEnabled)
+      _editableText.showToolbar();
+  }
+
+  bool _shouldShowHandles(SelectionChangedCause cause) {
+    // When the text field was triggered by anything other than a touch screen,
+    // we shouldn't show the selection handles.
```

Comment:
You're right. A pen (stylus) should also trigger the selection overlay because it's still interacts through the touch screen. Do you think `invertedStylus` should also trigger the selection overlay? I don't have a pen at the moment to try this.